### PR TITLE
Ilc typescript support

### PR DESCRIPTION
**Pull request by blackrabbit99**
ILC supports typescript v1

### CODE REVIEW

File: ilc/tsconfig.client.json
Comment (stas-nc) [inline ilc/tsconfig.client.json:1]: What about extending some base config in the root folder?

File: ilc/build/webpack.js [hunk 1]
```diff
@@ -19,6 +19,16 @@ module.exports = {
     mode: 'production',
     module: {
         rules: [
+            {
+                test: /\.(js|ts)$/,
+                use: {
+                    loader: 'ts-loader',
```
Comment (b1ff) [inline ilc/build/webpack.js:25]: no any rust? :(

File: ilc/package.json [hunk 2]
```diff
@@ -4,15 +4,16 @@
   "description": "",
   "main": "server/index.js",
   "scripts": {
-    "test": "cross-env NODE_ENV=test mocha",
+    "test": "npm run build:server && cross-env NODE_ENV=test mocha",
     "test:ci": "npm run test -- --forbid-only --exit",
     "test:watch": "npm run test -- --watch",
```
Comment (b1ff) [inline ilc/package.json:9]: I assume now this command won't work. Maybe better `ts-node`?

**Review by b1ff — CHANGES_REQUESTED** [review]

File: ilc/package.json (hunk 2 above)
Comment (blackrabbit99) [reply ilc/package.json:9]: this works. Idea is source require conf file like list of plugins so the easiest way to run mocha tests is to run build and execute tests from dist

File: ilc/tsconfig.client.json
Comment (blackrabbit99) [reply ilc/tsconfig.client.json:1]: That is good idea. Will add base config

File: ilc/package.json (hunk 2 above)
Comment (b1ff) [reply ilc/package.json:9]: can you explain how it work in `--watch` mode, if I change typescript file and tests are re-run?

Comment (blackrabbit99) [reply ilc/package.json:9]: Watch mode in such setup won't work. I got your point.

**Review by b1ff — APPROVED** [review]

File: ilc/server/serveStatic.js [hunk 3]
```diff
@@ -7,11 +7,11 @@ module.exports = function (isProduction) {
         return serveStatic('public');
     }
 
-    require('../systemjs/build');
+    require('../../systemjs/build');
```
Comment (yehor-manzhula) [inline ilc/server/serveStatic.js:10]: Why the path changed?

**Review by yehor-manzhula — APPROVED** [review]

**Pull request merged**